### PR TITLE
fix: sanitize email HTML templates

### PR DESCRIPTION
**Pull request by joeyorlando**
## Which issue(s) this PR closes

Closes https://github.com/grafana/oncall-private/issues/2760

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: engine/requirements.in [hunk 1]
```diff
@@ -68,3 +68,4 @@ google-auth-oauthlib==1.2.0
 # requests==2.32.3 is installing idna==3.6 but supports idna>=2.5,<4
 # https://github.com/psf/requests/blob/v2.32.3/setup.py#L63
 idna==3.7
+bleach==6.1.0
```
Comment (joeyorlando) [inline engine/requirements.in:71]: I didn't see any existing libraries/methods for sanitizing inputs so decided to go with this

**Comment by mderynck** [comment]
Web render does this [here](https://github.com/grafana/oncall/blob/e9d94ebc1ebc36d82a33f5ef7a5890474de6640b/engine/apps/alerts/incident_appearance/templaters/web_templater.py#L16) using a util method.  Maybe [email](https://github.com/grafana/oncall/blob/e9d94ebc1ebc36d82a33f5ef7a5890474de6640b/engine/apps/email/alert_rendering.py#L9) one should do similar and avoid a new dependency?

**Comment by mderynck** [comment]
Also I think it needs to be on post-format in case a template is written that will result in html but not recognized as such when it is in template form.

**Comment by joeyorlando** [comment]
@mderynck good point, thanks for pointing that out, updated 👍

**Comment by mderynck** [comment]
I'm not sure what the expected link behavior is for emails.  In the web render there is some additional code that converts plain links into <a> tags and handles markdown.  Do we need to pull that out of the web render for message and make it a common function?

**Comment by joeyorlando** [comment]
okay so I decided to just take a simpler approach (imo 😄) and just handle sanitizing this input in the `direct_paging` function in `apps.alerts.paging` (the API, public API, and Slack direct paging workflows all converge on this method):

**Before**
Screenshot 2024-10-02 at 11 07 42 AM


**After**
Screenshot 2024-10-02 at 11 06 23 AM

**Review by vadimkerr — APPROVED** [review]
lgtm

**Review by mderynck — APPROVED** [review]

**Pull request merged**